pkg/api/repository.go

```diff
@@ -11,6 +11,11 @@ type RepositoryResponse struct {
 	OrgID                string   `json:"org_id" readonly:"true"`              // Organization ID of the owner
 }
 
+// RepositoryResponseJSON wraps RepositoryResponse so the response can return as an empty object instead of nil
+type RepositoryResponseJSON struct {
+	Response *RepositoryResponse
+}
```

Comment:
If we don't want to use this, I think the alternative would be a RepositoryResponse object that would return all the fields with their zero values. (mostly empty string)

Reply:
Does this actually work with a python binding?  The generated json doesn't seem to indicate that the value can be null

Reply:
(and i tried something similar in my validation pr, as i wanted null responses if 'name' or 'url' wasn't provided, but couldn't actually get it to work with the generated python api bindings :(

Reply:
The value wouldn't be null it would be `{}`

Reply:
Not sure how to check the generated python api bindings

Reply:
run this to generate it in the output/ directory
```
 podman run --rm -v .:/project:z docker.io/openapitools/openapi-generator-cli generate -i /project/api/openapi.json -g python -o  /project/output/
```

then:
```
cd ./output
```

and then run python.  You can use this script to get you started (this is what i used to test the validation api, you'll need to change it to use your api):

```
import openapi_client 
from openapi_client.api import  repositories_api 
from openapi_client.model.api_repository_validation_request import ApiRepositoryValidationRequest


configuration = openapi_client.Configuration(
    host = "http://localhost:8000/api/content_sources/v1.0"
)

# Enter a context with an instance of the API client
api_client = openapi_client.ApiClient(configuration=configuration, header_name="x-rh-identity", header_value="eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6IjEiLCJpbnRlcm5hbCI6eyJvcmdfaWQiOiIxIn19fQo=")
api_instance = repositories_api.RepositoriesApi(api_client)


api_repository_request = [ ApiRepositoryValidationRequest(  
    name="name_example",
    url="http://redhat.com/", ),ApiRepositoryValidationRequest()]

api_response = api_instance.validate_repository_parameters(api_repository_request)
```

Reply:
actually, in this case i think it works.  (reading the api code) its because the bindings only would return null for Repository  in the case of an error, and technically the object we're returning only actually is returned on success.  The error response is not mapped to an object within the schema.

Reply:
in which case, i don't think this part is actaully needed to be changed.  At the core problem, there's no difference in returning: 
```
{
  repository: nil}
}
```
and
```
{
  repository: {
   result: nil
  }
}
```
i don't think?

Reply:
Oh, yeah. What I pushed is not what I meant to push.

Reply:
I removed the "Response" field which the linter doesn't like, but it makes the output `repository: {}`. I still need to try generating the python bindings and seeing if this helps. I assume we need remove `null` from all responses though.

Reply:
i see what you're saying that it does produce {}, but do you know why?  It doesn't make much sense to me.  was there some docs around this you found?

I wonder if something like "BulkCreatedRepository" would be a better name than RepositoryResponseJSON?

Reply:
I saw it on a stackoverflow I can't find, but there was no explanation anyway.

My understanding is by wrapping it, then json is marshalling a struct with nothing in it, so it gives `{}`. Where as previously it would marshal a `*RepositoryResponse = nil`, and give `null`. In short, I guess it's `{nil}` vs `nil`.

You could accomplish something similar with a struct where all the fields have the `omitempty` tag. If all the fields are omitted, then `{}` is the result. But here we need RepositoryResponse to return the empty fields in some cases, so that's not an option.

Reply:
It's weird. That's why I'm not sure that we want to use it.

Reply:
I'm going to try overriding the marshal function and see where that gets me